src/ctia/stores/es/crud.clj

```diff
@@ -78,25 +80,50 @@
                     (build-create-result model coerce-fn)))
                 (remove-es-actions items) models)}))
 
+
+(s/defn get-docs-with-indices
+  "Retrieves a documents from a search \"ids\" query. It enables to retrieves documents from an alias that points to multiple indices. It returns the documents with full hits meta data including the real index in which is stored the document."
+  [{:keys [conn index]} :- ESConnState
+   mapping :- s/Keyword
+   ids :- [s/Str]
+   params]
+  (let [ids-query (q/ids (map ensure-document-id ids))
+        res (d/query conn
+                     index
+                     (name mapping)
+                     ids-query
+                     (assoc (make-es-read-params params)
+                            :full-hits?
+                            true))]
+    (:data res)))
+
+(s/defn get-doc-with-index
+  "Retrieves a document from a search \"ids\" query. It is used to perform a get query on an alias that points to multiple indices. It returns the document with full hits meta data including the real index in which is stored the document."
+  [conn-state :- ESConnState
+   mapping :- s/Keyword
+   _id :- s/Str
+   params]
+  (first (get-docs-with-indices conn-state mapping [_id] params)))
+
 (defn handle-create
   "Generate an ES create handler using some mapping and schema"
   [mapping Model]
   (let [coerce! (coerce-to-fn (s/maybe Model))]
     (s/fn :- (s/maybe [Model])
-      [state :- ESConnState
+      [{:keys [index props] :as state} :- ESConnState
        models :- [Model]
        ident
        {:keys [refresh] :as params}]
       (try
-        (map #(build-create-result % coerce!)
-             (d/bulk-create-doc (:conn state)
-                                (map #(assoc %
-                                             :_id (:id %)
-                                             :_index (:index state)
-                                             :_type (name mapping))
-                                     models)
-                                (or refresh
-                                    (get-in state [:props :refresh] false))))
+        (mapv #(build-create-result % coerce!)
```

Comment:
I think you use `mapv` here just for better message output. If this is the case then I think you should revert them back to `map`. Looking at the code it is not clear that `map` if faster/better than `mapv` but  it looks like this is the case.

Reply:
No it's just a testimony of me fighting with a bug that I thought was related to lazyness :-). I remove it. Thx.